### PR TITLE
Release Spanner v0.30.0

### PR DESCRIPTION
**Pull request by jonparrott**

### CODE REVIEW

**Comment by jonparrott** [comment]
Must be rebased on #4730 before merging.

File: spanner/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,24 @@
 
 [1]: https://pypi.org/project/google-cloud-spanner/#history
 
+## 0.30.0
+
+### Implementation Changes
+
+- The underlying autogenerated client library was re-generated to pick up new features and resolve bugs. (#4695)
```
Comment (dhermes) [inline spanner/CHANGELOG.md]: 80 char limit or meh? (I have tried to uphold this. There is also the "semantic line breaks" approach.)

File: spanner/CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,24 @@
...
+- Made sure **exactly** one of `start_*`/`end_*` are passed to KeyRange. (#4618)
+- Made `row`, `consume_all`, and `consume_next` private (#4492)
```
Comment (dhermes) [inline spanner/CHANGELOG.md]: Probably worth mentioning on which class they are private?

File: spanner/CHANGELOG.md [hunk 3]
```diff
@@ -4,6 +4,24 @@
...
+
+### Documentation
+
+- Brought Spanner README more in line with others. (#4306, #4317)
+
+### Testing
```
Comment (dhermes) [inline spanner/CHANGELOG.md]: Worth putting in release notes?

**Comment by tseaver** [comment]
I still believe that the validation change in the `KeyRange` constructor in #4618 was ill-advised (see #4694).  If we leave it in place for this release, it should be highlighted as a breaking change.

**Comment by dhermes** [comment]
@tseaver I was going to mention that in my review, thanks for pointing it out.

**Comment by tseaver** [comment]
#4492 is a breaking change, as well: it *removed* the `rows` property and the `consume_all` method, making `consume_next` private.

**Comment by jonparrott** [comment]
Reverting it. We can revisit it at a later time if needed.

File: spanner/CHANGELOG.md (hunk 2 above)
Comment (jonparrott) [reply spanner/CHANGELOG.md]: Done (commit will appear during rebase)

File: spanner/CHANGELOG.md (hunk 3 above)
Comment (jonparrott) [reply spanner/CHANGELOG.md]: Yep.

File: spanner/CHANGELOG.md (hunk 1 above)
Comment (jonparrott) [reply spanner/CHANGELOG.md]: Done (commit will appear during rebase)

**Comment by jonparrott** [comment]
Rebased and good to merge when CI is green and api_core 0.1.4 is released (#4730)

**Comment by chemelnucfin** [comment]
@jonparrott That change was suggested by @vkedia

**Comment by jonparrott** [comment]
@chemelnucfin two different breaking changes were discussed, which were you referring to? The change of making a few member private was *not* reverted and I'm totally cool with it.

**Comment by chemelnucfin** [comment]
The few member private ones.  _row and _consume_next.

sorry for not being clear.  Trying to deal with the spanner keys issue
right now.

On Wed, Jan 10, 2018 at 5:01 PM, Jon Wayne Parrott <notifications@github.com

**Comment by jonparrott** [comment]
#4735 is now included, thanks, @chemelnucfin.

Waiting for CI and then merging. :)

**Pull request merged**